server/remote_asset/fetch_server/fetch_server_test.go

```diff
@@ -215,67 +222,77 @@ func TestFetchBlobWithCache(t *testing.T) {
 			}, "", resource.CacheType_CAS, tc.storageFunc).ToProto())
 			require.NoError(t, err)
 			require.True(t, exist)
+
+			exist, err = te.GetCache().Contains(ctx, digest.NewResourceName(&repb.Digest{
+				Hash:      contentDigest.GetHash(),
+				SizeBytes: resp.GetBlobDigest().GetSizeBytes(),
+			}, "", resource.CacheType_CAS, tc.checksumFunc).ToProto())
+			require.NoError(t, err)
+			require.True(t, exist)
```

Comment:
I don't think we have really tested storing digests in the cache other than blake3 or sha256, so this might open us up to some new behavior that we should test more IMO. I would be fine supporting SHA512 etc. for validation purposes but when storing in the cache I'm not sure we want to start supporting new digest funcs just yet.

Reply:
done.

But what should I do to test more IMO. ( maybe in next pr )

Reply:
The current PR added validations for other checksum types without writing the blob to cache as well as improved our existing test setup. The double-writes to the cache have been removed.

I think this is good enough for merging. We will discuss adding support for caching SHA512/SHA384 blobs internally and decide whether to add it in a separate PR.